### PR TITLE
Initialize RequestParser & Implementation

### PR DESCRIPTION
**Pull request by AriaZhou**

### CODE REVIEW

File: src/RequestParser.cpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    case '{':
+    case '}':
+    case ' ':
+    case '\t':
+      return true;
+    default:
+      return false;
+  }
+}
+
```
Comment (LinRenHong) [inline src/RequestParser.cpp:65]: Maybe you can add comment to show that what result is expected

File: src/include/RequestParser.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    new_line_3,
+  } state;
+
+  // Temporary storage of header name
+  std::string header_name_tmp;
+  // Temporary storage of header value
+  std::string header_value_tmp;
+};
+
+#endif  //_REQUESTPARSER_H
```
Comment (LinRenHong) [inline src/include/RequestParser.h:94]: Please add newline at the end of file.

File: src/RequestParser.cpp [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+      } else {
+        return fail;
+      }
+    case new_line_3:
+      return (input == '\n') ? success : fail;
+    // TODO: parse body
+    default:
+      return fail;
+  }
+}
```
Comment (Jazz86) [inline src/RequestParser.cpp]: please add newline

**Review by kungyao — APPROVED** [review]
Viewed

**Review by john0312 — APPROVED** [review]
Please add unit test in this PR (preferably) or in another PR.

**Review by LinRenHong — APPROVED** [review]

File: src/RequestParser.cpp (hunk 1 above)
Comment (john0312) [reply src/RequestParser.cpp:65]: This is a reasonable review.

File: src/test/TestRequestParser.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      "Content-Type: text/plain; "
+      "charset=utf-8\r\n"
+      "Content-Length: 4\r\n"
+      "Connection: close\r\n"
+      "\r\n"
+      "test";
+
+  RequestParser::parse_result result;
+  std::tie(result, std::ignore) =
+      parser.Parse(&packet, test_request, test_request + 20);
```
Comment (john0312) [inline src/test/TestRequestParser.h]: Consider making this:
`sizeof(test_request)-kTestRemainingSize`

**Review by john0312 — APPROVED** [review]
Approved assuming the comments are addressed.

**Pull request merged**